### PR TITLE
HDF5 2.0.0 updates

### PR DESCRIPTION
**Pull request by brtnfld**
HDF5 2.0.0 API updates, including the renaming of H5VLstart_lib_state and H5VLfinish_lib_state, along with a new context state.

### CODE REVIEW

File: src/h5_async_vol.c [hunk 1]
```diff
@@ -5271,8 +5250,8 @@ async_attr_create_fn(void *foo)
 
     // Restore previous library state
     assert(task->h5_state);
-    if (H5VLstart_lib_state() < 0) {
-        fprintf(fout_g, "  [      ABT ERROR] %s H5VLstart_lib_state failed\n", __func__);
+    if (H5VLOPEN_LIB_CONTEXT(&task->h5_context) < 0) {
+        fprintf(fout_g, "  [      ABT ERROR] %s %s failed\n", __func__, FUNC_H5VLOPEN_LIB_CONTEXT);
         goto done;
     }
```
Comment (brtnfld) [inline src/h5_async_vol.c:5290]: This pattern of changes is repeated for the other 2.0 changes.

File: src/h5_async_vol.c [hunk 2]
```diff
@@ -5319,8 +5298,8 @@ async_attr_create_fn(void *foo)
     func_log(__func__, "execute success");
 
 done:
-    if (is_lib_state_restored && H5VLfinish_lib_state() < 0)
-        fprintf(fout_g, "  [      ABT ERROR] %s H5VLfinish_lib_state failed\n", __func__);
+    if (is_lib_state_restored && H5VLCLOSE_LIB_CONTEXT(task->h5_context) < 0)
+        fprintf(fout_g, "  [      ABT ERROR] %s %s failed\n", __func__, FUNC_H5VLCLOSE_LIB_CONTEXT);
     if (NULL != task->h5_state && H5VLfree_lib_state(task->h5_state) < 0)
```
Comment (brtnfld) [inline src/h5_async_vol.c:5337]: This pattern of changes is repeated for the other 2.0 changes.

File: src/h5_async_vol.c [hunk 3]
```diff
@@ -8938,52 +8918,81 @@ async_dataset_read_fn(void *foo)
     assert(task->async_obj->obj_mutex);
     assert(task->async_obj->magic == ASYNC_MAGIC);
     while (1) {
```
Comment (brtnfld) [inline src/h5_async_vol.c]: The diff looks confusing from here on; only #ifdef for HDF5 1.13 code was removed.

Comment (brtnfld) [reply src/h5_async_vol.c]: Will do 1.13 changes in another PR.

**Review by houjun — APPROVED** [review]

**Review by jhendersonHDF — APPROVED** [review]

**Pull request merged**